bletl/parsing/blpro.py

```diff
@@ -407,38 +407,42 @@ def transform_into_filtertimeseries(
         key = None
         times = None
         values = None
-        if fs.filter_type == "Intensity" and ("Biomass" in fs.filter_name or "BS" in fs.filter_name):
-            key = f"BS{int(fs.gain_1)}"
-            times = measurements.xs(filter_number, level="filterset")["time"].unstack()
-            values = measurements.xs(filter_number, level="filterset")["amp_ref_1"].unstack()
-        elif fs.filter_type in {"pH", "DO"} and not return_uncalibrated_optode_data:
-            key = fs.filter_type
-            times = measurements.xs(filter_number, level="filterset")["time"].unstack()
-            values = measurements.xs(filter_number, level="filterset")["cal"].unstack()
-        elif fs.filter_type in {"pH", "DO"} and return_uncalibrated_optode_data:
-            key = fs.filter_type
-            times = measurements.xs(filter_number, level="filterset")["time"].unstack()
-            values = measurements.xs(filter_number, level="filterset")["phase"].unstack()
-        elif fs.filter_type == "Intensity":
-            key = fs.filter_name
-            times = measurements.xs(filter_number, level="filterset")["time"].unstack()
-            values = measurements.xs(filter_number, level="filterset")["amp_ref_1"].unstack()
-        else:
+        if filter_number in measurements.index.get_level_values("filterset"): #test if any filterset is not available in measurements due to invalid data issue24
```

Comment:
Two things:
1. Place comments on their own line to maintain civilized line lengths
2. If you invert the condition (`if filter_number not in ...`) you can directly do the `loggern.warn and continue` part instead of having it 20 lines further below. This keeps the logically related code close together, and you won't have to indent the other four `if/elif/elif/elif/else` parts.

If you like to learn more about inverting if statements check this short video: https://www.youtube.com/watch?v=8YUw3f3BdOI

Reply:
I think I have done all your proposals :-) Please check if this is right :-) Thank you!